src/Nancy/Responses/EmbeddedFileResponse.cs

```diff
@@ -47,19 +48,17 @@ private Stream GetResourceContent(Assembly assembly, string resourcePath, string
         {
             var resourceName = assembly
                 .GetManifestResourceNames()
-                .Where(x => GetFileNameFromResourceName(resourcePath, x).Equals(name, StringComparison.OrdinalIgnoreCase))
-                .Select(x => GetFileNameFromResourceName(resourcePath, x))
-                .FirstOrDefault();
+                .FirstOrDefault(x => GetFileNameFromResourceName(resourcePath, x).Equals(name, StringComparison.OrdinalIgnoreCase));

```

Comment:
There is still a select here (I know this is not specific to the PR, just an observation)